virttest/utils_misc.py

```diff
@@ -3987,3 +3989,34 @@ def setup_remote(self):
         logging.debug("To check remote boolean value: %s", boolean_curr)
         if boolean_curr != self.remote_bool_value:
             raise exceptions.TestFail(result.stderr.strip())
+
+
+def get_model_features(model_name):
+    """
+    /usr/share/libvirt/cpu_map.xml defines all CPU models.
+    One CPU model is a set of features.
+    This function is to get features of one specific model.
+
+    :params model_name: CPU model name, valid name is given in cpu_map.xml
+    :return: feature list, like ['apic', 'ss']
+
+    """
+    features = []
+    conf = "/usr/share/libvirt/cpu_map.xml"
+    if not os.path.isfile(conf):
+        logging.warn("Configuration file is not found or not file: %s" % conf)
+        return []
+
+    output = process.run("cat %s" % conf)
+    root = ET.fromstring(output.stdout.strip())
+    try:
+        for model_node in root.findall('arch/model'):
+            if model_node.get('name') == model_name:
+                for feature in model_node.findall('feature'):
+                    features.append(feature.get('name'))
+    except ET.ParseError, error:
+        logging.warn("Configuration file %s has wrong xml format" % conf)
+    except AttributeError, elem_attr:
+        logging.warn("No attribute %s in file %s" % (str(elem_attr), conf))
+    finally:
```

Comment:
suggest not use return statement in finally block, so you may just remove finally statement

Reply:
Adopt. Thanks!